File: apps/client/types/workspace.ts
import z from "zod";
import { ScopeEnum } from "./enums";

export const WorkspaceSchema = z.object({
  id: z.uuid(),
  name: z.string().nonempty(),
  ownerId: z.uuid(),
  orgId: z.uuid().nullable(),
  scope: ScopeEnum,
  createdAt: z.date(),
  updatedAt: z.date(),
});

export type Workspace = z.infer<typeof WorkspaceSchema>;

File: apps/server/internal/models/workspace.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type WorkspaceScope string

const (
	WorkspacePersonal WorkspaceScope = "PERSONAL"
	WorkspaceOrg      WorkspaceScope = "ORG"
)

type Workspace struct {
	ID        uuid.UUID      `json:"id" db:"id"`
	OrgID     uuid.UUID      `json:"orgId" db:"org_id"`
	OwnerID   uuid.UUID      `json:"ownerId" db:"owner_id"`
	Scope     WorkspaceScope `json:"scope" db:"scope"`
	CreatedAt time.Time      `json:"createdAt" db:"created_at"`
	UpdatedAt time.Time      `json:"updatedAt" db:"updated_at"`
}

func (w WorkspaceScope) IsWorkspaceScopeValid() bool {
	switch w {
	case WorkspaceOrg, WorkspacePersonal:
		return true
	}
	return false
}
